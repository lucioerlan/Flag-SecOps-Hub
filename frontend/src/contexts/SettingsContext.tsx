import { storeSettings } from '@/utils/settings'
import { createContext, useState, ReactNode } from 'react'

const SettingsContext = createContext({
  settings: {
    token: '',
    email: '',
    isLoggedIn: false
  },
  setSettings: (settings: object) => {
    settings
  }
})

const defaultSettings = {
  token: '',
  email: '',
  isLoggedIn: false
}

export function SettingsProvider({
  settings,
  children
}: {
  settings: {
    token: string
    email: string
    isLoggedIn: boolean
  }
  children: ReactNode
}) {
  const [currentSettings, setCurrentSettings] = useState(settings || defaultSettings)

  const handleSaveSettings = (updatedSettings = {}) => {
    const mergedSettings = { ...currentSettings, ...updatedSettings }

    setCurrentSettings(mergedSettings)
    storeSettings(mergedSettings)
  }

  return (
    <SettingsContext.Provider
      value={{
        settings: currentSettings,
        setSettings: handleSaveSettings
      }}
    >
      {children}
    </SettingsContext.Provider>
  )
}

export const SettingsConsumer = SettingsContext.Consumer

export default SettingsContext
