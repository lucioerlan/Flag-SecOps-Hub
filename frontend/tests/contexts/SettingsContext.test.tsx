import { SettingsProvider } from '@/contexts/SettingsContext'
import { render } from '@testing-library/react'

describe('SettingsContext', () => {
  it('Should render SettingsContext', () => {
    const settings = {
      accessToken: '',
      isLoggedIn: false
    }

    const { container } = render(
      <SettingsProvider settings={settings}>
        <div>SettingsContext</div>
      </SettingsProvider>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
