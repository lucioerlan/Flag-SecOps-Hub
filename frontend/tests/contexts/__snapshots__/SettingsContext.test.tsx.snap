// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsContext Should render SettingsContext 1`] = `
<div>
  SettingsContext
</div>
`;
